spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/DeployerSocketUtils.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.spi.local;
+
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.util.Random;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import javax.net.ServerSocketFactory;
+
+import org.springframework.util.Assert;
+
+/**
+ * Simple utility methods for working with network sockets &mdash; for example, for
+ * finding available ports on {@code localhost}.
+ *
+ * <p>
+ * Within this class, a TCP port refers to a port for a {@link ServerSocket}; whereas, a
```

Comment:
This looks like a lift/shift from the now removed `SocketUtils`. If so, we should probably state that somewhere in the comments. Otherwise, the copyrights/authors list raises an eyebrow. 

Also, the original reason for it being [removed from SF](https://github.com/spring-projects/spring-framework/issues/28052)

> However, these utilities make no guarantee about the subsequent availability of a given port and are therefore unreliable.

although it has been reliable for our uses so far, correct?